### ISSUE TITLE
Add XSRF protection

### ISSUE THREAD

**Issue by djwells** [IdentityServer/IdentityServer3.Admin#14]
Add XSRF header protection to the API and update the UI to send the correct header.

**Comment by brockallen** [comment]
Are we using cookies or tokens for API calls?

**Comment by djwells** [comment]
In the past I have worked with custom HTTP headers for XSRF.  But I was going to look at how the main IdentityServer handles this before starting...it looks like you are passing a token value via cookies and doing some DataProtector stuff.

What is your recommendation on which approach to take?  We could base it on the AntiRequestForgeryToken and related middleware from IdSvr.

**Comment by brockallen** [comment]
Well, it's not needed if you use a bearer token for API calls. But that does mean you need some sort of way to obtain a token. I recently pushed these to bower:

**Comment by brockallen** [comment]
Also, I'll be updating IdentityManager soon to solve this exact same problem -- the admin tool and IdMgr will most likely use the same security models -- either local requests are allowed or tokens are used.

**Comment by djwells** [comment]
This brings up a general point:  there are common interfaces/code that we probably want to share between the two projects.  Related to the Owin/WebApi pipeline, I'm thinking of XSRF, Require SSL, Logging, Data Protection, Dependency Injection, etc.  At some point we should all talk through what bits belong in common and a strategy to share those among the projects.  Going through the IdentityServerOptions class is probably a good place to start to compile such a list.

**Comment by ManuelRauber** [comment]
@djwells @brockallen If sharing makes sense, I would create a new issue for that, where it can be discussed?

**Comment by brockallen** [comment]
Sure. I'm working on it today in IdMgr -- I hope to be done by end of day.
